src/components/switch/__internal__/switch-slider.style.ts

```diff
@@ -30,103 +30,108 @@ const StyledSwitchSlider = styled.span`
     height: 28px;
     left: 0;
     letter-spacing: 1px;
-    position: absolute;
+    position: relative;
     text-transform: uppercase;
     top: 0;
-    width: 64px;
-    min-width: fit-content;
+    width: 100%
     z-index: 2;
-    border-radius: ${theme?.roundedCornersOptOut
-      ? "90px"
-      : "var(--borderRadius400)"};
+    border-radius: ${
+      theme?.roundedCornersOptOut ? "90px" : "var(--borderRadius400)"
+    };
     border-style: solid;
     border-color: var(--colorsActionMinor400);
     border-width: var(--borderWidth200);
     box-sizing: border-box;
+    margin-top: ${size === "large" ? "-47px" : "-28px"};
```

Comment:
comment (non-blocking): I think this is needed as we're setting the height on 3 elements. We can probably get rid of this if we were to only apply the height to `StyledSwitchSlider` when size is large and also setting the input to have the following:

```
input {
    position: absolute;
    height: fit-content;
  }